## Exploratory Data Analysis
- Load in your csv.gz's of results for each year extracted.
- Concatenate the data into 1 dataframe for the remainder of the analysis.
- Once you have your data from the API, they would like you to perform some light EDA to show:
-1 How many movies had at least some valid financial information (values > 0 for budget OR revenue)?
-2 Please exclude any movies with 0's for budget AND revenue from the remaining visualizations.
-3 How many movies are there in each of the certification categories (G/PG/PG-13/R)?
-4 What is the average revenue per certification category?
-5 What is the average budget per certification category?


In [1]:
import json
import tmdbsimple as tmdb
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from tqdm.notebook import tqdm_notebook

In [2]:
# Open saved file and preview again
df_00 = pd.read_csv("Data/final_tmdb_data_2000.csv.gz", low_memory = False)
df_00.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.112     2008.0            PG  

[5 rows x 26 columns]

In [3]:
# Open saved file and preview again
df_01 = pd.read_csv("Data/final_tmdb_data_2001.csv.gz", low_memory = False)
df_01.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0035423    0.0  /ab5yL8zgRotrICzGbEl10z24N71.jpg                   NaN   
2  tt0114447    0.0                               NaN                   NaN   
3  tt0118589    0.0  /9NZAirJahVilTiDNCHLFcdkwkiy.jpg                   NaN   
4  tt0118652    0.0  /mWxJEFRMvkG4UItYJkRDMgWQ08Y.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  48000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 14, 'nam...      NaN   
2         0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 28, 'n...      NaN   
3  22000000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...      NaN   
4   1000000.0  [{'id': 27, 'name': 'Horror'}, {'id': 9648, 'n...      NaN   

         id original_language         original_title  ...     revenue  \
0       NaN               NaN                    NaN  ...         NaN   
1   11232.0                en         Kate & Leopold  ...  76019048.0   
2  151007.0                en       The Silent Force  ...         0.0   
3   10696.0                en                Glitter  ...   5271666.0   
4   17140.0                en  The Attic Expeditions  ...         0.0   

   runtime                                   spoken_languages    status  \
0      NaN                                                NaN       NaN   
1    118.0  [{'english_name': 'Italian', 'iso_639_1': 'it'...  Released   
2     90.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3    104.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4    100.0  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                  title  \
0                                                NaN                    NaN   
1  If they lived in the same century they'd be pe...         Kate & Leopold   
2  They left him for dead... They should have fin...       The Silent Force   
3   In music she found her dream, her love, herself.                Glitter   
4  His search for peace of mind... will leave his...  The Attic Expeditions   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         6.318     1146.0         PG-13  
2    0.0         5.000        3.0           NaN  
3    0.0         4.600      118.0         PG-13  
4    0.0         5.100       28.0             R  

[5 rows x 26 columns]

In [5]:
df_concat = pd.concat([df_00,df_01])

In [6]:
df_concat.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.112     2008.0            PG  

[5 rows x 26 columns]

In [7]:
df_00.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1208 entries, 0 to 1207
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1208 non-null   object 
 1   adult                  1207 non-null   float64
 2   backdrop_path          644 non-null    object 
 3   belongs_to_collection  110 non-null    object 
 4   budget                 1207 non-null   float64
 5   genres                 1207 non-null   object 
 6   homepage               68 non-null     object 
 7   id                     1207 non-null   float64
 8   original_language      1207 non-null   object 
 9   original_title         1207 non-null   object 
 10  overview               1185 non-null   object 
 11  popularity             1207 non-null   float64
 12  poster_path            1073 non-null   object 
 13  production_companies   1207 non-null   object 
 14  production_countries   1207 non-null   object 
 15  rele

In [8]:
df_01.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1293 entries, 0 to 1292
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1293 non-null   object 
 1   adult                  1292 non-null   float64
 2   backdrop_path          664 non-null    object 
 3   belongs_to_collection  86 non-null     object 
 4   budget                 1292 non-null   float64
 5   genres                 1292 non-null   object 
 6   homepage               107 non-null    object 
 7   id                     1292 non-null   float64
 8   original_language      1292 non-null   object 
 9   original_title         1292 non-null   object 
 10  overview               1263 non-null   object 
 11  popularity             1292 non-null   float64
 12  poster_path            1142 non-null   object 
 13  production_companies   1292 non-null   object 
 14  production_countries   1292 non-null   object 
 15  rele

In [9]:
budg_no_0 = df_concat['budget'] > 0

In [10]:
rev_no_0 = df_concat['revenue'] > 0

In [11]:
budg_rev_no_0 = df_concat.loc[budg_no_0 | rev_no_0]

In [12]:
budg_rev_no_0.head()

imdb_id  adult                     backdrop_path  \
1   tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg   
4   tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg   
6   tt0119273    0.0  /f5C03doOWiauu37bToKXtpgP5bS.jpg   
10  tt0120263    0.0  /ynXVuylP8upazjz8lrqb1PEMkdR.jpg   
11  tt0120467    0.0  /knok3mNReKqPTplnnqz7E4dd7mD.jpg   

                                belongs_to_collection      budget  \
1                                                 NaN  10000000.0   
4                                                 NaN    150000.0   
6   {'id': 141086, 'name': 'Heavy Metal Collection...  15000000.0   
10                                                NaN         0.0   
11                                                NaN    120000.0   

                                               genres  \
1   [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
4   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
6   [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
10  [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...   
11  [{'id': 18, 'name': 'Drama'}, {'id': 27, 'name...   

                                homepage       id original_language  \
1                                    NaN  62127.0                en   
4   http://www.wkw-inthemoodforlove.com/    843.0                cn   
6                                    NaN  16225.0                en   
10                                   NaN  34070.0                sv   
11                                   NaN  19085.0                en   

                original_title  ...     revenue  runtime  \
1              The Fantasticks  ...         0.0     86.0   
4                         花樣年華  ...  12854953.0     99.0   
6             Heavy Metal 2000  ...         0.0     88.0   
10  Sånger från andra våningen  ...     80334.0     98.0   
11                      Vulgar  ...     14904.0     87.0   

                                     spoken_languages    status  \
1   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
4   [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
6   [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
10  [{'english_name': 'Russian', 'iso_639_1': 'ru'...  Released   
11  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                              tagline  \
1       Try to remember the first time magic happened   
4   Feel the heat, keep the feeling burning, let t...   
6   not to survive the fight in the external world...   
10                                                NaN   
11   Everyone loves a clown... some more than others.   

                          title  video  vote_average vote_count certification  
1               The Fantasticks    0.0         5.500       22.0           NaN  
4          In the Mood for Love    0.0         8.112     2008.0            PG  
6              Heavy Metal 2000    0.0         6.134      160.0             R  
10  Songs from the Second Floor    0.0         7.245      220.0           NaN  
11                       Vulgar    0.0         5.400       42.0             R  

[5 rows x 26 columns]

In [14]:
budg_rev_no_0.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 623 entries, 1 to 1272
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                623 non-null    object 
 1   adult                  623 non-null    float64
 2   backdrop_path          523 non-null    object 
 3   belongs_to_collection  103 non-null    object 
 4   budget                 623 non-null    float64
 5   genres                 623 non-null    object 
 6   homepage               80 non-null     object 
 7   id                     623 non-null    float64
 8   original_language      623 non-null    object 
 9   original_title         623 non-null    object 
 10  overview               620 non-null    object 
 11  popularity             623 non-null    float64
 12  poster_path            601 non-null    object 
 13  production_companies   623 non-null    object 
 14  production_countries   623 non-null    object 
 15  relea

#### - There are 623 entries that have at least a budget or a revenue of over $0

In [15]:
df_concat.groupby('certification')['budget'].describe()

count          mean           std  min  25%         50%  \
certification                                                            
-                1.0  0.000000e+00           NaN  0.0  0.0         0.0   
G               24.0  2.383333e+07  3.310283e+07  0.0  0.0  13000000.0   
NC-17            6.0  0.000000e+00  0.000000e+00  0.0  0.0         0.0   
NR              68.0  1.575001e+06  7.506164e+06  0.0  0.0         0.0   
PG              64.0  2.459766e+07  3.818498e+07  0.0  0.0     75002.0   
PG-13          180.0  3.128282e+07  3.398642e+07  0.0  0.0  22000000.0   
R              452.0  9.998904e+06  1.985928e+07  0.0  0.0         0.0   
Unrated          1.0  0.000000e+00           NaN  0.0  0.0         0.0   

                      75%          max  
certification                           
-                     0.0          0.0  
G              31250000.0  115000000.0  
NC-17                 0.0          0.0  
NR                    0.0   60000000.0  
PG             35750000.0  127500000.0  
PG-13          50000000.0  140000000.0  
R              10000000.0  115000000.0  
Unrated               0.0          0.0

#### How many movies in each certification category:

- There are 24 movies with Ratings G
- There are 64 movies with Ratings PG
- There are 180 movies with Ratings PG-13
- There are 452 movies with Ratings R

#### What is average budget for each certification

- Budget of G = 22,052,060
- Budget of PG = 23,496,270
- Budget of PG-13 = 31,062,680
- Budget of R = 9,823,285

In [16]:
df_concat.groupby('certification')['revenue'].describe()

count          mean           std  min  25%         50%  \
certification                                                            
-                1.0  0.000000e+00           NaN  0.0  0.0         0.0   
G               24.0  7.216332e+07  1.283376e+08  0.0  0.0  19191331.5   
NC-17            6.0  0.000000e+00  0.000000e+00  0.0  0.0         0.0   
NR              68.0  2.221905e+06  8.247379e+06  0.0  0.0         0.0   
PG              64.0  6.143535e+07  1.504771e+08  0.0  0.0         0.5   
PG-13          180.0  7.224140e+07  1.237283e+08  0.0  0.0  19522730.0   
R              452.0  1.682113e+07  4.864767e+07  0.0  0.0         0.0   
Unrated          1.0  0.000000e+00           NaN  0.0  0.0         0.0   

                       75%          max  
certification                            
-                     0.00          0.0  
G              86524956.75  579707738.0  
NC-17                 0.00          0.0  
NR                    0.00   47386287.0  
PG             40273195.50  976475550.0  
PG-13          86444674.50  871368364.0  
R               6115683.75  465361176.0  
Unrated               0.00          0.0

#### What is average revenue for each certification

- Revenue of G = 72,163,320
- Revenue of PG = 61,435,350
- Revenue of PG-13 = 72,241,400
- Revenue of R = 16,821,130

In [17]:
## Save current dataframe 
df_concat.to_csv("Data/tmdb_results_combined.csv.gz",compression='gzip',index=False)
df_concat_test = pd.read_csv("Data/tmdb_results_combined.csv.gz", low_memory = False)
df_concat_test.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres  \
0         NaN                                                NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   

                               homepage        id original_language  \
0                                   NaN       NaN               NaN   
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   

    original_title  ...     revenue  runtime  \
0              NaN  ...         NaN      NaN   
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.112     2008.0            PG  

[5 rows x 26 columns]